src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -268,7 +269,7 @@ public class KNNSettings {
     /**
      * This setting identifies KNN index.
      */
-    public static final Setting<Boolean> IS_KNN_INDEX_SETTING = Setting.boolSetting(KNN_INDEX, false, IndexScope);
```

Comment:
Can you check what behavior happens for an index created on an old version of OpenSearch? Is the setting immutable after upgrading? This could be checked by adding a test in https://github.com/opensearch-project/k-NN/tree/main/qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc